### ISSUE TITLE
White lines between colored cells when using partial vline or hline

### ISSUE THREAD

**Issue by jtl5770** [lvjr/tabularray#215]
Using texlive 2021, and tabularray 2022A.

My MWE

```
\documentclass[11pt]{article}
\usepackage{xcolor}
\usepackage{tabularray}

\definecolor{colA}{rgb}{0.8,0.9,0.8}
\definecolor{colB}{rgb}{0.5,0.7,0.5}
\definecolor{colC}{rgb}{0.95,0.85,0.6}
\definecolor{colD}{rgb}{0.9,0.5,0.5}

\begin{document}
\begin{tblr}{
    column{1} = {colA}, column{2} = {colB}, column{3} = {colC},
    cell{1}{1,3} = {colD}, cell{2}{3} = {colD}, cell{3}{2} = {colD},
    hline{1} = {1,3}{solid}, hline{2} = {1}{solid}, hline{3,4} = {2}{solid}, hline{3} = {3}{solid},
    vline{1,2} = {1}{solid}, vline{4} = {1,2}{solid}, vline{2} = {3}{solid}, vline{3}, 
    %%%%%%%% Note the ugly white lines.
    %%%%%%%% I would like to NOT have to do the following to fix this (but instead it being "right" automatically):
    % hline{2} = {2}{colB, leftpos=0, rightpos=0}, hline{2} = {3}{colD, leftpos=0, rightpos=0}, hline{3} = {1}{colA},
    % vline{2} = {2}{colB},
  }
  Alpha & Beta & Gamma \\
  One   & Two  & Three \\
  Tic   & Tac  & Toe
\end{tblr}
\end{document}
```

This yields the following result.

![tabularraytest](https://user-images.githubusercontent.com/24967370/158844033-88d9c8b6-bd72-4763-9cfe-f4e133425127.gif)

See the comments in the MWE what I do to fix this - basically it boils down to having to insert rules colored like the background of the cells and inserting leftpos/rightpos=0 in strategic places. So it is possible, but it seems incredibly awkward having to do this in the first place. Ideally this would be handled in a way that these white lines are not created in the first place.

How could this be done? I have no idea how to do it in a current model where the vline/hline takes up space on its own next to the cells. But - could we implement vlines/hlines (maybe with their own new name and semantic) that actually don't take up space, but are rather printed on top? Half of the width of the rule would then be added to the cell right (or top in the case of a hline) and half of it to the cell left (or bottom), so the visible part of the cell would still be exactly the same as it would be without the rule in place. This would ensure that the cells are always directly stacked next  to each other and the background wouldn't have any discontinuities.

**Comment by lvjr** [comment]
In the future there will be a `tikz` library (see #29) and you can draw borders on the cells with `tikz` commands.

**Comment by jtl5770** [comment]
Using `tikz` seems like the obvious answer, but isn't it a bit like cheating :-) ?
For the second half of my suggestion (the automatic enlargement of the cell with padding) - could this be implemented in the `tikz` case at all?

**Comment by jtl5770** [comment]
Actually, for vertical lines I wonder if it would be ok to just use something like \clap to make them take up no space... but I didn't find a way to insert it for vlines. And I have no idea if something similar exists to smash vertical space, too. Sorry, but my TeX knowledge is very basic, so I don't think I can implement this myself without help :-)

**Comment by lvjr** [comment]
I haven't decided how and when to handle this. But at lease I don't think enlarging cell sizes is a good idea. It will probably break other things.

**Comment by jtl5770** [comment]
Yes, I understand that. If needed, I assume this can be easily done manually, by choosing e.g. rowsep accordingly.
This will be much less of a problem than having to hunt the white lines and filling them with colored v/hlines manually.

**Comment by lvjr** [comment]
> Actually, for vertical lines I wonder if it would be ok to just use something like \clap to make them take up no space... but I didn't find a way to insert it for vlines. And I have no idea if something similar exists to smash vertical space, too. Sorry, but my TeX knowledge is very basic, so I don't think I can implement this myself without help :-)

In general, it is doable to make hlines and vlines take up no space. But I think we only need to support single hline/vline, as double hlines/vlines look weird. Also don't expect the new feature will arrive soon.
